src/main/kotlin/org/rust/ide/formatter/RustAlignmentStrategy.kt

```diff
@@ -0,0 +1,42 @@
+package org.rust.ide.formatter
+
+import com.intellij.formatting.Alignment
+import com.intellij.lang.ASTNode
+
+/**
+ * Mimics [com.intellij.formatting.alignment.AlignmentStrategy], but offers more flexible API
+ */
+interface RustAlignmentStrategy {
```

Comment:
I have allowed myself to "fork" `AlignmentStrategy`, the overall idea is very nice, but IntelliJ's implementation is terrible for our use.